### PR TITLE
skip lintrunner install on Windows.

### PR DESCRIPTION
**Pull request by xuhancn**
`lintrunner` is not support Windows x64. Ref: https://pypi.org/project/lintrunner/#files 

When we install python dependency by `pip install -r requirements.txt` on Windows x64, it will failed on `lintrunner`.
image

Solution: skip install `lintrunner` on Windows.
Reference doc: https://peps.python.org/pep-0508/#environment-markers 

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ filelock
 networkx
 jinja2
 fsspec
-lintrunner
+lintrunner ; platform_system != "Windows" or (platform_system == "Windows" and platform_machine == "x86")
```
Comment (albanD) [inline requirements.txt:21]:
```suggestion
lintrunner ; platform_system != "Windows"
```

I don't think there is any plan to support it on any windows version unfortunately.

Comment (xuhancn) [reply requirements.txt]: @albanD okay, let's skip Windows all version.

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**